libsupport/src/ProgressTracer.cpp

```diff
@@ -204,6 +209,22 @@ katana::ProgressSpan::LogError(
                 {"error.context", fmt::format("{}", error)}});
 }
 
+void
+katana::ProgressSpan::LogProfile() {
+#ifdef KATANA_USE_JEMALLOC
+  // Write a profile according to the environment variable
+  // MALLOC_CONF
+  int ret = mallctl("prof.dump", NULL, NULL, NULL, 0);
+  if (ret) {
+    KATANA_LOG_ERROR("jemalloc dump: {}", std::strerror(ret));
+    return;
+  }
+  static int stage;
+  SetTags({{"profile_stage", stage}});
```

Comment:
I would make this a log since it would be better if developers don't write to the same tag on a span multiple times